common/scripts/copyright-linter.js

```diff
@@ -6,41 +6,50 @@ const fs = require("fs");
 const path = require("path");
 const child_process = require("child_process");
 
-// Get all arguments after the positional argument indicator, "--"
-const filePaths = process.argv.reduce((acc, cur) => {
-  if (acc) {
-    acc.push(cur);
-    return acc;
-  } else if (cur === "--") {
-    return [];
-  } else if (cur === "--fix") {
-    // Support manually updating to fix changes made before the linter was fixed.
-    return child_process.execSync("git diff --name-only master")
-      .toString()
-      .split("\n")
-      .map(f => path.join(__dirname, "../..", f))
-      .filter(f => /\.(js|ts|tsx|scss|css)$/.test(f));
-  }
-}, false);
+function getFileNames(lintBranch) {
+  // Get name of every file changed between current branch and master,
```

Comment:
Can this use just the diff between the last commit instead of using a specific branch? That way it will be applicable to all branches, not just ones targeting the master branch.

Reply:
This is pulled from the current linter where it by default lints the past commit but will lint relative to master with the optional parameter "--fix". This new version is similar but uses the optional parameter "--branch" to specify this case.

Would it be better to have the option to lint all files instead of the changes relative to master? Current performance tests show it managing all 4400 files in repo in 11 seconds which suggests covering ~20,000-24,000 files a minute.

Reply:
I changed this so the two options are to lint all changes in the last commit (default), or supply --all to lint every file in the repo since the performance difference is negligible.